lib/core/diagnostics.js

```diff
@@ -114,6 +123,108 @@ if (undiciDebugLog.enabled || fetchDebuglog.enabled) {
   isClientSet = true
 }
 
+// Track fetch requests
+if (fetchDebuglog.enabled) {
+  const debuglog = fetchDebuglog
+
+  tracingChannels.start.subscribe(evt => {
+    const {
+      input
+    } = evt
+    debuglog(
+      'fetch has started request to %s',
+      input
+    )
+  })
+
+  tracingChannels.end.subscribe(evt => {
+    const {
+      input
+    } = evt
+    debuglog(
+      'fetch has received response from %s',
+      input
+    )
+  })
+
+  tracingChannels.asyncStart.subscribe(evt => {
+    const {
+      input,
+      result,
+      error
+    } = evt
+    if (result && error) {
+      debuglog(
+        'fetch has received response for %s - HTTP %d, error is %s',
+        input,
+        result.status,
+        error.message
+      )
+    } else if (result) {
+      debuglog(
+        'fetch has received response for %s - HTTP %d',
+        input,
+        result.status
+      )
+    } else if (error) {
+      debuglog(
+        'fetch has errored for %s - %s',
+        input,
+        error.message
+      )
+    } else {
+      debuglog(
+        'fetch has started request to %s',
+        input
+      )
+    }
+  })
+
+  tracingChannels.asyncEnd.subscribe(evt => {
+    const {
+      input,
+      result,
+      error
+    } = evt
+    if (result && error) {
+      debuglog(
+        'fetch has received response for %s - HTTP %d, error is %s',
+        input,
+        result.status,
+        error.message
+      )
+    } else if (result) {
+      debuglog(
+        'fetch has received response for %s - HTTP %d',
+        input,
+        result.status
+      )
+    } else if (error) {
+      debuglog(
+        'fetch has errored for %s - %s',
+        input,
+        error.message
+      )
+    } else {
+      debuglog(
+        'fetch has ended request to %s',
+        input
+      )
+    }
+  })
```

Comment:
The `asyncEnd` is undifferentiated from `asyncStart` here and should probably be omitted as it would look like the thing happened twice.

Reply:
removing